### ISSUE TITLE
Toggle autoplay in ConcatenatingAudioSource

### ISSUE THREAD

**Issue by martinc2907** [ryanheise/just_audio#917]
**Is your feature request related to a problem? Please describe.**
The ConcatenatingAudioSource automatically skips to the next item when the previous item finishes playback. But I need a way to control this behavior by toggling 'autoplay' on/off.

**Describe the solution you'd like**
A way to toggle autoplay would be nice. 

**Describe alternatives you've considered**
I have tried to manually pause the player when an item is nearing the end of its duration. This solution is not optimal as the pause method is async and playback may finish while the pause method is being processed. It is also not possible to pause at exactly the end of the audio. 

**Additional context**

**Comment by ryanheise** [comment]
I think it's a good idea. While there is a native API to achieve this on iOS, there isn't an API for this yet on Android.

There is an ExoPlayer issue tracking this at https://github.com/google/ExoPlayer/issues/5660

**Comment by martinc2907** [comment]
Ah I see, surprised to see that ExoPlayer does not have this functionality yet.

**Comment by ryanheise** [comment]
Note that you can in theory emulate this behaviour quite accurately by using `ConcatenatingAudioSource` as a tool to prepare the gapless playback queue rather than using it directly as your playlist. You can dynamically remove all following items in the queue if you want it to stop at the end of the current item. Then you can dynamically add more items when you want to allow it to autoplay.

**Comment by martinc2907** [comment]
Sounds like an interesting solution as long as an accurate queue(i.e. items after the current index) does not need to be maintained. My UI needs to show the next items in the queue so I don't think it would work.

**Comment by ryanheise** [comment]
Yes, the solution should definitely work but the key point is to use this API as an auto playback queue rather than a playlist. Once you do that, you should be aware that just_audio's state represents the state of that queue and does not represent the playlist state of your app. But accepting that, there is nothing preventing you from maintaining your own state for such purposes and only using just_audio to manage the auto playback queue.

**Comment by martinc2907** [comment]
> I think it's a good idea. While there is a native API to achieve this on iOS, there isn't an API for this yet on Android.
> 
> There is an ExoPlayer issue tracking this at [google/ExoPlayer#5660](https://github.com/google/ExoPlayer/issues/5660)

Seems like ExoPlayer supports [pausing at the end of media items](https://github.com/google/ExoPlayer/issues/5660#issuecomment-714497693) since v2.12. What it doesn't support is pausing at arbitrary frames.

**Comment by martinc2907** [comment]
https://developer.android.com/reference/androidx/media3/exoplayer/ExoPlayer#setPauseAtEndOfMediaItems(boolean)
This is the method they use.

**Comment by ryanheise** [comment]
Thanks for noticing that. This issue therefore has no blockers - aside from just the fact that someone still needs to implement it ;-)

**Comment by BraveEvidence** [comment]
This will help https://www.youtube.com/watch?v=IMQdSTlTXjA

**Comment by 0xharkirat** [comment]
Hey @ryanheise @martinc2907 , Can you guys please give me a short example of pausing or stopping the player after current song in playlist is finished.

**Comment by ryanheise** [comment]
See: https://stackoverflow.com/questions/73339177/flutter-with-just-audio-is-it-possible-to-create-a-playlist-that-doesnt-aut

**Comment by Chaphasilor** [comment]
Is there any other way to perform an action before the player skips to the next `AudioSource`? I basically need to hook into the skipping process in order to update the queue before the actual skip happens, but still need gapless playback for when the queue doesn't change (which is the default). Overriding `skipToNext()` works for skips caused through user action, but it doesn't cover the automatic skipping :/

**Comment by ryanheise** [comment]
@Chaphasilor this issue is a feature request to be able to toggle autoplay. It sounds like you aren't commenting on this feature request but just have a general question. The place to do this is on StackOverflow, although note that you should always search to see if a question has been asked and answered before posting a new one. I will hide your comment and mine with the "off topic" tag.

**Comment by whitecodel** [comment]
This has been resolved in PR https://github.com/ryanheise/just_audio/pull/1500. 

The new `autoPlayNextPlayListItem` parameter in `AudioLoadConfiguration` allows you to control whether the player automatically advances to the next playlist item or pauses at the end of each item.

**Usage:**
```dart
final player = AudioPlayer(
  audioLoadConfiguration: AudioLoadConfiguration(
    autoPlayNextPlayListItem: false, // Pause at end of each item
  ),
);
